piprot/piprot.py

```diff
@@ -263,7 +263,7 @@ def main(
     session = FuturesSession()
     results = []
 
-    for req, version, ignore in requirements:
+    for req, version, ignore in list(set(requirements)):
```

Comment:
@SamuelMS doing this doesn't guarantee that the order will stay the same - unfortunately that means that verbatim output may end up in the incorrect order.

Perhaps a solution where you check to see if the requirement is in the results already, then skip adding it would work instead.